### ISSUE TITLE
Test Timeouts in ReplicatorTest

### ISSUE THREAD

**Issue by merlimat** [apache/incubator-pulsar#1138]
I have been seeing a lot of CI builds timing out after 100 minutes, all were blocked on the `ReplicatorTest`. For example: 

https://builds.apache.org/job/pulsar-pull-request/1321/console

```
[INFO] Running org.apache.pulsar.broker.service.SubscriptionSeekTest
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 7.457 s - in org.apache.pulsar.broker.service.SubscriptionSeekTest
[INFO] Running org.apache.pulsar.broker.service.ReplicatorTest
Build timed out (after 100 minutes). Marking the build as aborted.

```

**Comment by merlimat** [comment]
After adding timeout in the setup/teardown phases: 

https://builds.apache.org/job/pulsar-pull-request/1349/

```
org.testng.internal.thread.ThreadTimeoutException: Method org.apache.pulsar.broker.service.ReplicatorTest.testReplicatorProducerClosing() didn't finish within the time-out 30000
	at java.lang.Object.wait(Native Method)
	at java.lang.Thread.join(Thread.java:1260)
	at io.netty.util.HashedWheelTimer.stop(HashedWheelTimer.java:380)
	at org.apache.pulsar.client.impl.PulsarClientImpl.shutdown(PulsarClientImpl.java:409)
	at org.apache.pulsar.broker.service.BrokerService.lambda$2(BrokerService.java:360)
	at org.apache.pulsar.broker.service.BrokerService$$Lambda$380/1989638644.accept(Unknown Source)
	at org.apache.pulsar.common.util.collections.ConcurrentOpenHashMap$Section.forEach(ConcurrentOpenHashMap.java:386)
	at org.apache.pulsar.common.util.collections.ConcurrentOpenHashMap.forEach(ConcurrentOpenHashMap.java:160)
	at org.apache.pulsar.broker.service.BrokerService.close(BrokerService.java:358)
	at org.apache.pulsar.broker.PulsarService.close(PulsarService.java:170)
	at org.apache.pulsar.broker.service.ReplicatorTest.testReplicatorProducerClosing(ReplicatorTest.java:679)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)

```


```
org.testng.internal.thread.ThreadTimeoutException: Method org.apache.pulsar.broker.service.ReplicatorTest.shutdown() didn't finish within the time-out 30000
	at sun.misc.Unsafe.park(Native Method)
	at java.util.concurrent.locks.LockSupport.park(LockSupport.java:175)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.parkAndCheckInterrupt(AbstractQueuedSynchronizer.java:836)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquireQueued(AbstractQueuedSynchronizer.java:870)
	at java.util.concurrent.locks.AbstractQueuedSynchronizer.acquire(AbstractQueuedSynchronizer.java:1199)
	at java.util.concurrent.locks.ReentrantLock$NonfairSync.lock(ReentrantLock.java:209)
	at java.util.concurrent.locks.ReentrantLock.lock(ReentrantLock.java:285)
	at org.apache.pulsar.broker.PulsarService.close(PulsarService.java:156)
	at org.apache.pulsar.broker.service.ReplicatorTestBase.shutdown(ReplicatorTestBase.java:248)
	at org.apache.pulsar.broker.service.ReplicatorTest.shutdown(ReplicatorTest.java:92)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:54)
	at org.testng.internal.InvokeMethodRunnable.run(InvokeMethodRunnable.java:44)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
```

**Comment by sijie** [comment]
This is fixed by #1154 #1354
